authz.go

```diff
@@ -225,6 +226,20 @@ func (a *authorizer) authorize(ctx context.Context, sr authz.SubjectAccessReview
 			break
 		}
 	}
+
+	// if cache and dry run are enabled, error need to be added to log if authorize result from cache and authorize result from zts / zms are different.
+	if a.AuthorizationConfig.Config.UseCache {
+		// decision is nil, meaning it is not authorized through cache, but zts/zms authorized
+		if decision == nil && granted {
+			log.Printf("There is a mismatch between cache result and zms result. Cache granted: unauthorized, Athenz granted: authorized for principal: %s on check: %s", principal, via)
+		}
+
+		// decision is not empty, but result from zts/zms is not authorize
+		if decision != nil && decision.status.Allowed && !granted {
+			log.Printf("There is a mismatch between cache result and zms result. Athenz granted: unauthorized, Cache granted: authorized for principal: %s on check: %s", principal, decision.via)
```

Comment:
Can we keep both logs in the same format as the splunk alert so the alert matches both cases? The alert is specifically looking for true or false. Another option is updating the alert as well.
```
There is a mismatch between cache result and zms result | rex "(?P<cache_decision>true|false), Athenz granted: (?P<athenz_decision>true|false) for user: (?P<principal>[^ ]+) on check: (?P<action>[^ ]+) on (?P<athenz_domain>[^ ]+):(?P<resource>[^ ]+)" 
```

Reply:
k will update that.